### PR TITLE
Implementing S3 Server-Side Encryption

### PR DESCRIPTION
**Pull request by rvahudson**
This is my first PR and open source contribution, so please be gentle if I haven't met requirements for contribution. 

S3 has built-in server-side encryption that can be implemented via the API (docs [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/serv-side-encryption.html) and [here](http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property)). This PR implements SSE using the S3-managed key system, as well as user-managed keys via [KMS](http://docs.aws.amazon.com/AmazonS3/latest/dev/UsingKMSEncryption.html). I have not implemented Customer-Provided Keys (SSE-C) in this PR, but may at some point in the future.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -124,6 +130,20 @@ function S3Storage (opts) {
     case 'undefined': this.getStorageClass = defaultStorageClass; break
     default: throw new TypeError('Expected opts.storageClass to be undefined, string or function')
   }
+
+  switch (typeof opts.sse) {
```
Comment (LinusU) [inline index.js]: I would prefer it if this was called `serverSideEncryption`, just to map to the aws api

File: index.js [hunk 2]
```diff
@@ -124,6 +130,20 @@ function S3Storage (opts) {
...
+    case 'function': this.getSSE = opts.sse; break
+    case 'string': this.getSSE = staticValue(opts.sse); break
+    case 'undefined': this.getSSE = defaultSSE; break
+    default: throw new TypeError('Expected opts.sse to be undefined, string or function')
+  }
+
+  switch (typeof opts.sseKms) {
```
Comment (LinusU) [inline index.js]: Same here, I guess it would be `sseKmsKeyId`

**Comment by rvahudson** [comment]
@LinusU Keeping the naming in line with the S3 API makes sense. Suggested changes were made and pushed to this branch.

**Comment by ChrisAlvares** [comment]
@LinusU any chance this gets merged anytime soon?

**Comment by LinusU** [comment]
I feel that it's ready, let's give @badunk a few days to come with comments, otherwise I'll merge and release it

Feel free to remind me if I forget :)

**Comment by badunk** [comment]
thanks for the tests!

**Comment by rvahudson** [comment]
Awesome, glad I could help!

**Pull request merged**